integration_tests/suite/test_phonebook_http.py

```diff
@@ -358,3 +391,304 @@ def test_get_multi_tenant(self, sub, main):
+                    'lastname': 'McContact',
+                    'number': str(1000000000 + i),
+                }
+                for i in range(5000)
+            ],
+        )
+        assert not errors
+
+    @classmethod
+    def tearDownClass(self):
+        self.stack.close()
+        super().tearDownClass()
+
+    def test_get_all(self):
+        client = self.get_client(VALID_TOKEN_MAIN_TENANT)
+
+        with self.source(
+            client,
+            {
+                'phonebook_uuid': self.phonebook['uuid'],
+                'name': self.phonebook['name'] + "-source",
+            },
+        ) as source:
+            response = client.session().get(
+                url=f"{client.phonebook_source.base_url}/{source['uuid']}/contacts"
+            )
+            response.raise_for_status()
+            body = response.json()
+            assert_that(
+                body,
+                has_entries(
+                    total=len(self.contacts),
+                    filtered=len(self.contacts),
+                    items=all_of(
+                        has_length(len(self.contacts)),
+                        only_contains(
+                            has_entries(
+                                firstname=instance_of(str),
+                                lastname=instance_of(str),
+                                number=instance_of(str),
+                                id=instance_of(str),
+                            )
+                        ),
+                    ),
+                ),
+            )
+
+    def test_get_paginated(self):
+        client = self.get_client(VALID_TOKEN_MAIN_TENANT)
+
+        with self.source(
+            client,
+            {
+                'phonebook_uuid': self.phonebook['uuid'],
+                'name': self.phonebook['name'] + "-source",
+            },
+        ) as source:
+            limit = 100
+            contact_ids: set[str] = set()
+            for i in range(5000 // 100):
+                logger.debug("request %d", i)
+                response = client.session().get(
+                    url=f"{client.phonebook_source.base_url}/{source['uuid']}/contacts",
+                    params={'limit': 100, 'offset': i * limit},
+                )
+                response.raise_for_status()
+                body = response.json()
+                assert_that(
+                    body,
+                    has_entries(
+                        total=len(self.contacts),
+                        filtered=len(self.contacts),
+                        items=all_of(
+                            has_length(100),
+                            only_contains(
+                                has_entries(
+                                    firstname=instance_of(str),
+                                    lastname=instance_of(str),
+                                    number=instance_of(str),
+                                    id=instance_of(str),
```

Comment:
This test does not assert that we have the correct page: the test would pass if pages were ordered randomly